src/core/expression/predicate.visitor.spec.js

```diff
@@ -0,0 +1,239 @@
+			let foo = predicateVisitor.visit(group);
+			let res = foo(123);
+			expect(res).to.equal(true);
+		});
+
+		it('check for equality with op = "or"', () => {
+			let group = {
+				kind: 'group',
+				op: 'or',
+				right: {
+					kind: 'condition',
+					op: 'equals',
+					right: 'Some string',
+					left: 'value'
+				},
+				left: {
+					kind: 'condition',
+					op: 'equals',
+					right: 123,
+					left: 'value'
+				}
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(group);
+			let res = foo(123);
+			expect(res).to.equal(true);
+			expect(foo('Some string')).to.equal(true)
+		});
+	});
+
+	describe('visit condition', () => {
+		it('check for equality with op = "notEquals"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'notEquals',
+				right: 123,
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(999);
+			expect(res).to.equal(true);
+		});
+
+		it('check for equality with op = "greaterThan"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'greaterThan',
+				right: 123,
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(999);
+			expect(res).to.equal(true);
+		});
+
+		it('check for equality with op = "lessThan"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'lessThan',
+				right: 123,
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(1);
+			expect(res).to.equal(true);
+		});
+
+		it('check for equality with op = "between"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'between',
+				right: [10, 20],
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(15);
+			expect(res).to.equal(true);
+			let res2 = foo(17);
+			expect(res2).to.equal(true);
+			let res3 = foo(5);
+			expect(res3).to.equal(false);
+			let res4 = foo(21);
+			expect(res4).to.equal(false);
+		});
+
+		it('check for equality with op = "in"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'in',
+				right: [10, 20],
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo(10);
+			expect(res).to.equal(true);
+		});
+
+		it('check whether "findSomeString" contains "Some"', () => {
+			let condition = {
+				kind: 'condition',
+				op: 'like',
+				right: 'SOME',
+				left: 'value'
+			};
+			let predicateVisitor = new PredicateVisitor(valueFactory);
+			let foo = predicateVisitor.visit(condition);
+			let res = foo('findSomeString');
+			expect(res).to.equal(true);
+			let res2 = foo('findSomeString');
```

Comment:
Seems that you test the same again :) and there will be better to create own test for this case also